nevergrad/functions/gym/gym_multi.py

```diff
@@ -0,0 +1,592 @@
+    "FrozenLake-v0",
+    "FrozenLake8x8-v0",
+    "CliffWalking-v0",
+    "NChain-v0",
+    "Roulette-v0",
+    "Taxi-v3",
+    "CubeCrash-v0",
+    "CubeCrashSparse-v0",
+    "CubeCrashScreenBecomesBlack-v0",
+    "MemorizeDigits-v0",
+]
+
+
+# We do not use "conformant" which is not consistent with the rest.
+CONTROLLERS = [
+    "linear",  # Simple linear controller.
+    "neural",  # Simple neural controller.
+    "deep_neural",  # Deeper neural controller.
+    "semideep_neural",  # Deep, but not very deep.
+    "structured_neural",  # Structured optimization of a neural net.
+    "memory_neural",  # Uses a memory (i.e. recurrent net).
+    "deep_memory_neural",
+    "stackingmemory_neural",  # Uses a memory and stacks a heuristic and the memory as inputs.
+    "deep_stackingmemory_neural",
+    "semideep_stackingmemory_neural",
+    "extrapolatestackingmemory_neural",  # Same as stackingmemory_neural + suffix-based extrapolation.
+    "deep_extrapolatestackingmemory_neural",
+    "semideep_extrapolatestackingmemory_neural",
+    "semideep_memory_neural",
+    "multi_neural",  # One neural net per time step.
+    "noisy_neural",  # Do not start at 0 but at a random point.
+    "scrambled_neural",  # Why not perturbating the order of variables ?
+    "noisy_scrambled_neural",
+    "stochastic_conformant",  # Conformant planning, but still not deterministic.
+]
+
+
+NO_LENGTH = ["ANM", "Blackjack", "CliffWalking", "Cube", "Memorize", "ompiler"]
+
+
+# Environment used for CompilerGym.
+class SmallActionSpaceLlvmEnv(gym.ActionWrapper):
+    """A wrapper for the LLVM compiler environment that exposes a tiny subset of
+    the full discrete action space (the subset was hand pruned to contain a mix
+    of "good" and "bad" actions).
+    """
+
+    action_space_subset = [
+        "-break-crit-edges",
+        "-early-cse-memssa",
+        "-gvn-hoist",
+        "-gvn",
+        "-instcombine",
+        "-instsimplify",
+        "-jump-threading",
+        "-loop-reduce",
+        "-loop-rotate",
+        "-loop-versioning",
+        "-mem2reg",
+        "-newgvn",
+        "-reg2mem",
+        "-simplifycfg",
+        "-sroa",
+    ]
+
+    def __init__(self, env, flags=None):
+        super().__init__(env=env)
+        # Array for translating from this tiny action space to the action space of
+        # the wrapped environment.
+        self.true_action_indices = [self.action_space[f] for f in self.action_space_subset]
+
+    def action(self, action: Union[int, List[int]]):
+        if isinstance(action, int):
+            return self.true_action_indices[action]
+        else:
+            return [self.true_action_indices[a] for a in action]
+
+
+class CompilerGym(ExperimentFunction):
+    def __init__(self):
+        action_space_size = len(SmallActionSpaceLlvmEnv.action_space_subset)
+        parametrization = (
+            ng.p.Array(shape=(num_episode_steps,)).set_bounds(0, action_space_size - 1).set_integer_casting()
+        )
```

Comment:
you'd need several options of parametrizations here,
this one is fine though I would move bouds to -0.49 and action_space_size - 0.51 (hacky :s)
and `Choice(action_space_size, deterministic)` for both deterministic=True and False
and TransitionChoice(action_space_size)

Reply:
I've copied this code from Chris' code. Maybe you can discuss it in the chat with him as well.